### PR TITLE
[json-rpc] clean up TransactionDataView mapper and add tests

### PR DESCRIPTION
**Pull request by xli**
## Motivation

* Remove unnecessary error check in TransactionDataView#from(Transaction), we already handled all types
* Add more test cases to cover corner cases.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Unit test and integration test

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/libra/developers.libra.org, and link to your PR here.)

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: json-rpc/types/src/views.rs [hunk 1]
```diff
@@ -520,13 +520,11 @@ impl ScriptView {
...
-                t.into_inner().map(|x| TransactionDataView::BlockMetadata {
-                    timestamp_usecs: x.1,
-                })
-            }
-            Transaction::GenesisTransaction(_) => Ok(TransactionDataView::WriteSet {}),
+        match tx {
+            Transaction::BlockMetadata(t) => TransactionDataView::BlockMetadata {
+                timestamp_usecs: t.timestamp_usec(),
+            },
+            Transaction::GenesisTransaction(_) => TransactionDataView::WriteSet {},
```
Comment (thefallentree) [inline json-rpc/types/src/views.rs:527]: should we rename this too?

File: json-rpc/types/src/views.rs [hunk 2]
```diff
@@ -554,11 +552,9 @@ impl From<Transaction> for TransactionDataView {
                     script_hash,
                     script_bytes,
                     script: t.into_raw_transaction().into_payload().into(),
-                })
+                }
             }
-        };
-
-        x.unwrap_or(TransactionDataView::UnknownTransaction {})
```
Comment (thefallentree) [inline json-rpc/types/src/views.rs:561]: why were these removed ?  It is good for backward compatibility

File: json-rpc/types/src/views.rs (hunk 1 above)
Comment (xli) [reply json-rpc/types/src/views.rs:527]: GenesisTransaction(WriteSetPayload),

I think keep it as WriteSet is better, GenesisTransaction seems too specific, although I don't mind to rename, but will need add new type to TransactionDataView, maybe another diff if we prefer.

File: json-rpc/types/src/views.rs (hunk 2 above)
Comment (xli) [reply json-rpc/types/src/views.rs:561]: because we don't need it, there is no "unknown" transaction, you can see all transaction type is mapped.
it is possible in the future, we will have unknown one, but that will be breaking change to existing code.
in above code, you can see I remove "Ok", so there is no "unwrap_or" anymore.

Technically, I'd like to return error if unwrap failed, instead of return unknown transaction type, which hides real error.
But here we really don't have unknown transaction anymore.

**Review by thefallentree — APPROVED** [review]
